Vulkan: shrink PipelineKey from 608 bytes to 304 bytes, et al.

prideout opened the pull request:
This optimizes our memory footprint and does some cleanup. 
 There are no behavioral changes.

This PR is composed of 3 commits that should stay separate.

pixelflinger commented on filament/backend/src/vulkan/VulkanPipelineCache.h:
```diff
@@ -208,32 +205,63 @@ class VulkanPipelineCache : public CommandBufferObserver {
-        uint32_t padding1;                                                          // 4 bytes
+        VkShaderModule shaders[SHADER_MODULE_COUNT]; // 0
+        VkRenderPass renderPass;                     // 16
+        uint16_t topology : 16;                      // 24
```
i s the bitfield needed here, since it's a `uint16_t`?

pixelflinger replied:
I think it would be very useful to have the size of each field in comment as well

pixelflinger commented on filament/backend/src/vulkan/VulkanSwapChain.h:
```diff
@@ -38,19 +38,17 @@ struct VulkanSwapChain : public HwSwapChain {
-    VulkanTexture& getColorTexture() const;
 
-    // TODO: privatize more fields
+    VulkanTexture& getColorTexture() const;
+    VulkanTexture& getDepthTexture() const;
```
seems a bit "dangerous" to have a `const` method return a non-const reference.

pixelflinger commented on filament/backend/src/vulkan/VulkanSwapChain.cpp:
```diff
@@ -343,18 +347,12 @@ bool VulkanSwapChain::hasResized() const {
-VulkanTexture& VulkanSwapChain::getColorTexture() const {
-    return *this->mColor[this->currentSwapIndex];
+VulkanTexture& VulkanSwapChain::getDepthTexture() const {
+    return *mDepth;
```
should we worried that this could be `nullptr`?

prideout commented:
Thanks for the good feedback @pixelflinger , addressed all your comments except the last one, the swap chain is always backed by color & depth textures.
